src/pal/src/exception/seh.cpp

```diff
@@ -212,6 +212,12 @@ VOID
 SEHProcessException(PEXCEPTION_POINTERS pointers)
 {
     pointers->ContextRecord->ContextFlags |= CONTEXT_EXCEPTION_ACTIVE;
+#if defined(_ARM_) && defined(FEATURE_PAL)
+    // WinContext assumes that the least bit of Pc is always 1 (denoting thumb)
+    // although the pc value retrived from native context might not have set the least bit.
+    // This becomes especially problematic if the context is on the JIT_WRITEBARRIER.
+    pointers->ContextRecord->Pc |= 0x1;
```

Comment:
We should to do that in the CONTEXTFromNativeContext instead. That way, it would cover both activation injection and hardware exceptions. Also, no need to check for FEATURE_PAL in PAL.

It should be here:
https://github.com/dotnet/coreclr/blob/master/src/pal/src/thread/context.cpp#L491